cni/pkg/nodeagent/informers.go

```diff
@@ -103,8 +103,12 @@ func (s *InformerHandlers) GetPodIfAmbientEnabled(podName, podNamespace string)
 }
 
 func (s *InformerHandlers) Start() {
+	// Wait for all events to be queued
 	kube.WaitForCacheSync("informer", s.ctx.Done(), s.pods.HasSynced, s.namespaces.HasSynced)
 	go s.queue.Run(s.ctx.Done())
+	// Note that we are explicitly *not* doing
```

Comment:
Do we want to wait to get a connection before we call queue.Run then?

If we don't, I assume the impact is we might try to add, it fails, and we retry later? So probably not a big deal

Reply:
Yeah - we have `WithMaxRetries(5)` set up,  but we could exhaust those before  we get a connection, so let me experiment with delaying both snapshot creation and event processing until we get an initial connection.

Reply:
We can't delay event processing until we get a ztunnel connection (which would basically allow "enrolled" pods to fail open until we got a connection, which is bad - currently we fail closed (inject iptables rules, can't add to ztunnel, at least we fail closed).

But also (even before this PR) we might just lose events entirely if no ztunnel is available to handle them before we max out retry attempts, which doesn't seem great.

So what I did here was change event requeues - removing the cap on retry _attempts_ of failed events, but twiddle the queue retry timing.